scheduler-pool/benches/scheduler.rs

```diff
@@ -0,0 +1,894 @@
+                            .entry(**account)
+                            .and_modify(|indices: &mut Vec<usize>| indices.push(idx))
+                            .or_insert_with(|| vec![idx]);
+                    }
+                });
+
+            tx_account_locks
+                .iter()
+                .enumerate()
+                .map(|(idx, account_locks)| {
+                    let mut dep_graph: IntSet<usize> = IntSet::default();
+
+                    let readlock_conflict_accs = account_locks.writable.iter();
+                    let writelock_conflict_accs = account_locks
+                        .readonly
+                        .iter()
+                        .chain(account_locks.writable.iter());
+
+                    for acc in readlock_conflict_accs {
+                        if let Some(indices) = indices_read_locking_account.get(acc) {
+                            dep_graph.extend(indices.iter().take_while(|l_idx| **l_idx < idx));
+                        }
+                    }
+
+                    for acc in writelock_conflict_accs {
+                        if let Some(indices) = indicies_write_locking_account.get(acc) {
+                            dep_graph.extend(indices.iter().take_while(|l_idx| **l_idx < idx));
+                        }
+                    }
+                    dep_graph
+                })
+                .collect()
+        }
+
+        fn execute_batches(
+            &self,
+            bank: &Arc<Bank>,
+            pending_transactions: &[SanitizedTransaction],
+            receiver: &crossbeam_channel::Receiver<Signature>,
+        ) -> Result<()> {
+            if pending_transactions.is_empty() {
+                return Ok(());
+            }
+
+            let mut tx_account_locks: Vec<_> = Vec::with_capacity(pending_transactions.len());
+            for tx in pending_transactions {
+                tx_account_locks
+                    .push(tx.get_account_locks(bank.get_transaction_account_lock_limit())?);
+            }
+
+            // the dependency graph contains the indices that must be executed (marked with
+            // State::Done) before they can be executed
+            let dependency_graph = Self::build_dependency_graph(&tx_account_locks);
+
+            #[derive(Clone)]
+            enum State {
+                Blocked,
+                Processing,
+                Done,
+            }
+
+            let mut processing_states: Vec<State> = vec![State::Blocked; dependency_graph.len()];
+            let mut signature_indices: HashMap<&Signature, usize> =
+                HashMap::with_capacity(dependency_graph.len());
+            signature_indices.extend(
+                pending_transactions
+                    .iter()
+                    .enumerate()
+                    .map(|(idx, tx)| (tx.signature(), idx)),
+            );
+
+            let mut is_done = false;
+            while !is_done {
+                is_done = true;
```

Comment:
**minor**
As `is_done` is not used outside the loop, I think, it can be moved fully inside:

```suggestion
            loop {
                let mut is_done = true;
```